doc/cpp-style-guide.md

```diff
@@ -0,0 +1,933 @@
+
+In a few cases where the GSG give latitude we have made a choice.
+
+***Line length:***
+
+Each line of text in your code should be at most 120 characters long.
+
+#### Non-ASCII Characters
+
+Non-ASCII characters should be rare, and must use UTF-8 formatting.
+
+[link to GSG](https://google.github.io/styleguide/cppguide.html#Non-ASCII_Characters)
+
+#### Spaces vs. Tabs
+
+Use only spaces, and indent 2 spaces at a time.
+
+We use spaces for indentation. Do not use tabs in your code. You should set your editor to emit spaces when you hit
+the tab key.
+
+[link to GSG](https://google.github.io/styleguide/cppguide.html#Spaces_vs._Tabs)
+
+#### Function Declarations and Definitions
+
+Return type on the same line as function name, parameters on the same line if they fit. Wrap parameter lists which do not fit on a single line as you would wrap arguments in a function call.
+
+https://google.github.io/styleguide/cppguide.html#Function_Declarations_and_Definitions
+#### Lambda Expressions
+
+Format parameters and bodies as for any other function, and capture lists like other comma-separated lists.
+
+For by-reference captures, do not leave a space between the ampersand (&) and the variable name.
+
+[link to GSG](https://google.github.io/styleguide/cppguide.html#Formatting_Lambda_Expressions)
+
+#### Function Calls
+
+Either write the call all on a single line, wrap the arguments at the parenthesis, or start the arguments on a new line indented by four spaces and continue at that 4 space indent. In the absence of other considerations, use the minimum number of lines, including placing multiple arguments on each line where appropriate.
+
+[link to GSG](https://google.github.io/styleguide/cppguide.html#Function_Calls)
+
+#### Braced Initializer List Format
+
+Format a braced initializer list exactly like you would format a function call in its place.
+
+[link to GSG](https://google.github.io/styleguide/cppguide.html#Braced_Initializer_List_Format)
+
+#### Conditionals
+
+Prefer no spaces inside parentheses. The if and else keywords belong on separate lines.
+
+[link to GSG](https://google.github.io/styleguide/cppguide.html#Conditionals)
+
+#### Loops and Switch Statements
+
+Switch statements may use braces for blocks. Annotate non-trivial fall-through between cases. Braces are optional for single-statement loops. Empty loop bodies should use empty braces or continue.
+
+[link to GSG](https://google.github.io/styleguide/cppguide.html#Loops_and_Switch_Statements)
+
+#### Pointer and Reference Expressions
+
+No spaces around period or arrow. Pointer operators do not have trailing spaces. When declaring a pointer variable or
+argument, place the asterisk adjacent to the type:
+
+```C++
+// These are fine, space following.
+char* c;
+string const& str;
+
+// These are not fine, space preceding.
+char *c;
+string const &str;
+```
+
+[link to GSG](https://google.github.io/styleguide/cppguide.html#Pointer_and_Reference_Expressions)
+
+#### Boolean Expressions
```

Comment:
This is likely to be a polarizing choice and has very little effect on code quality, succinctness or correctness. In my opinion this falls under the category of a rule not worth remembering.

Reply:
You mean the boolean word operators, right?  I am clearly in the minority here.

Reply:
Didnt mean to make you feel alone :). Wanted to add one bit here - I felt the boolean word operator clause was slightly at odds with this goal "Be consistent with the broader C++ community as much as possible".

Reply:
Between you and @gamorris I think I am convinced.  Let's stick to the Google Style Guide on this too.